File: base_java/src/main/java/com/hesheng1024/base_java/ui/BaseActivityWithToolbar.java
package com.hesheng1024.base_java.ui;

import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.LayoutRes;
import androidx.annotation.NonNull;
import androidx.appcompat.widget.Toolbar;

import com.hesheng1024.base_java.R;
import com.hesheng1024.base_java.utils.LogUtil;

import java.util.Objects;

/**
 * @author hesheng1024
 * @date 2019/5/26 16:58
 */
public abstract class BaseActivityWithToolbar<P extends BasePresenter> extends BaseActivity<P> {
    private Toolbar mToolbar;
    private TextView mTitle;

    @Override
    protected View inflateContentView() {
        ViewGroup contentView = (ViewGroup) View.inflate(this, R.layout.activity_base, null);
        contentView.addView(initToolbar(contentView), 0);
        View.inflate(this, getLayoutId(), contentView);
        return contentView;
    }

    private View initToolbar(ViewGroup contentView) {
        mToolbar = (Toolbar) getLayoutInflater().inflate(R.layout.include_toolbar, contentView, false);
        if (isTranslucent()) {
            mToolbar.setPadding(0, getStatusBarHeight(), 0, 0);
        }
        mToolbar.setBackgroundResource(R.drawable.bg_toolbar);
        mToolbar.addView(getLayoutInflater().inflate(getToolbarContent(), mToolbar, false));
        setSupportActionBar(mToolbar);
        Objects.requireNonNull(getSupportActionBar()).setDisplayShowHomeEnabled(false);
        initToolbarEvent(mToolbar);
        return mToolbar;
    }

    private void initToolbarEvent(@NonNull Toolbar toolbar) {
        ImageView ivLeft = toolbar.findViewById(R.id.iv_toolbar_left);
        mTitle = toolbar.findViewById(R.id.tv_toolbar_title);
        if (ivLeft != null) {
            ivLeft.setOnClickListener(v -> finish());
        } else {
            LogUtil.e("ivLeft is null, maybe toolbar layout is diff or null");
        }
    }

    protected void setTheTitle(@NonNull String title) {
        if (mTitle != null) {
            mTitle.setText(title);
        } else {
            LogUtil.e("mTitle is null, maybe toolbar layout is diff or null");
        }
    }

    protected void setTheTitle(int resId) {
       setTheTitle(getResources().getText(resId).toString());
    }

    public Toolbar getToolbar() {
        return mToolbar;
    }

    @LayoutRes
    protected int getToolbarContent() {
        return R.layout.include_toolbar_content;
    }
}

File: base_kotlin/src/main/java/com/hesheng1024/base_kotlin/ui/BaseActivityWithToolbar.kt
package com.hesheng1024.base.base

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.widget.AppCompatImageView
import androidx.appcompat.widget.AppCompatTextView
import androidx.appcompat.widget.Toolbar
import com.hesheng1024.base_kotlin.R

/**
 *
 * @author hesheng1024
 * @date 2020/2/8 11:00
 */
abstract class BaseActivityWithToolbar<out P : BasePresenter<IBaseView, IBaseModel>> : BaseActivity<P>() {

    private var mToolbar: Toolbar? = null
    private var mTitle: AppCompatTextView? = null

    @SuppressLint("InflateParams")
    override fun inflateContentView(): View {
        val contentView = LayoutInflater.from(this).inflate(R.layout.activity_base, null)
        (contentView as ViewGroup).addView(initToolbar(contentView), 0)
        LayoutInflater.from(this).inflate(getLayoutId(), contentView as ViewGroup?)
        return contentView
    }

    private fun initToolbar(contentView: ViewGroup): View? {
        mToolbar = layoutInflater.inflate(R.layout.include_toolbar, contentView, false) as Toolbar?
        mToolbar?.let { toolbar ->
            if (isTranslucent()) {
                toolbar.setPadding(0, getStatusBarHeight(), 0, 0)
            }
            toolbar.setBackgroundResource(getStatusBarBackground())
            toolbar.addView(layoutInflater.inflate(getToolbarLayout(), mToolbar, false))
            setSupportActionBar(toolbar)
            supportActionBar?.setDisplayHomeAsUpEnabled(false)
            initToolbarEvent(toolbar)
        }
        return mToolbar
    }

    private fun initToolbarEvent(toolbar: Toolbar) {
        val ivLeft = toolbar.findViewById<AppCompatImageView>(R.id.iv_toolbar_left)
        mTitle = toolbar.findViewById(R.id.tv_toolbar_title)
        ivLeft?.setOnClickListener { finish() }
    }

    protected fun setTheTitle(title: String) {
        mTitle?.text = title
    }

    protected fun setTheTitle(resId: Int) {
        setTheTitle(resources.getText(resId).toString())
    }

    protected open fun getToolbarLayout(): Int = R.layout.include_toolbar_content

    protected fun getToolbar(): Toolbar? = mToolbar
}